src/services/timer.ts

```diff
@@ -1,4 +1,124 @@
-import app from "server";
-import { Server } from "socket.io";
-import QuizModel from "@models/quiz/quizModel"
-import http from 'http';
+import mongoose from 'mongoose'
+import QuizModel from '@models/quiz/quizModel'
+import logger from '@utils/logger'
+import { IQuiz, IParticipant } from '@types/quiz'
+enum QuizCode {
+  JoinQuiz = 'joinQuiz',
+  LeftQuiz = 'leftQuiz',
+  ServerDisconnect = 'server namespace disconnect',
+}
+
+const isQuizAcceptingAnswers = async (quizId: string) => {
+  const quiz: IQuiz = await QuizModel.findById(quizId)
+  if (!quiz) {
+    return false
+  }
+  if (!quiz.isPublished || !quiz.isAcceptingAnswers) {
+    return false
+  }
+  return true
+}
+
+const isParticipantGivingQuiz = async (quizId: string, userId: string) => {
+  const quiz: Iquiz = await QuizModel.findById(quizId)
+  if (!quiz) {
+    return false
+  }
+  const user: IParticipant = quiz.participants.find((participant) => {
+    if (participant.user.equals(mongoose.Types.ObjectId(userId))) {
+      return participant
+    }
+  })
+
+  let reason: string
+  if (!user) {
+    return false
+  }
+  if (user.isGivingQuiz || user.submitted) {
+    if (user.isGivingQuiz) reason = 'User is giving quiz in another portal'
+    if (user.submitted) reason = 'User has submitted the quiz'
+    return false
+  }
+  return true
+}
+
+async function checkUserQuizStatus(quizId: unknown, userId: unknown) {
+  const isAcceptingAnswers: unknown = await isQuizAcceptingAnswers(quizId)
+  const isGivingQuiz: unknown = await isParticipantGivingQuiz(quizId, userId)
+  if (!isAcceptingAnswers || !isGivingQuiz) {
```

Comment:
```suggestion
  if (!isAcceptingAnswers || isGivingQuiz) {
```
change this. if user is giving quiz then we should return false